insights/parsers/dse_ldif.py

```diff
@@ -0,0 +1,194 @@
+"""
+RHDSConfig - file ``/etc/dirsrv/slapd-<>/dse.ldif``
+===================================================
+
+The ``dseLdif`` module provides parsing for the
+``dse.ldif`` file.  The ``dseLdifConfig`` class
+implements the parsing and provides a ``list`` of all
+configuration lines present in the file.
+
+Sample input is provided in the *Examples*.
+
+Examples:
+    >>> dse_ldif_input = '''
+    ...
+    ... dn:
+    ... objectClass: top
+    ... aci: (targetattr != "aci")(version 3.0; aci "rootdse anon read access"; allow(
+    ...  read,search,compare) userdn="ldap:///anyone";)
+    ... creatorsName: cn=server,cn=plugins,cn=config
+    ... modifiersName: cn=server,cn=plugins,cn=config
+    ... createTimestamp: 20151021124922Z
+    ... modifyTimestamp: 20151021124922Z
+    >>> from insights.tests import context_wrap
+    >>> shared = {dseLdifConfig: dseLdifConfig(context_wrap(dse_ldif_input))}
+    >>> dse_ldif_config = shared[dseLdifConfig]
+    >>> 'objectClass' in dse_ldif_config
+    True
+    >>> 'creatorsName' in dse_ldif_config
+    True
+    >>> dse_ldif_config.get_line('creatorsName')
+    'creatorsName: cn=server,cn=plugins,cn=config'
+    >>> dse_ldif_config.get_values('modifyTimestamp')
+    ['20151021124922Z']
+    >>> dse_ldif_config.last('createTimestamp')
+    '20151021124922Z'    
+"""
+from collections import namedtuple
+from .. import Parser, parser, get_active_lines
+from insights.specs import Specs
+import re
+
+# optional whitespace, at least one non-whitespace (the keyword), at least one whitespace (space), a plus literal, anything
+PLUS_PATTERN = re.compile(r'^\s*\S+\s+\+.*$')
+
+
+@parser(Specs.dse_ldif)
+class dseLdifConfig(Parser):
```

Comment:
Class names should use CamelCase scheme.

Reply:
DSELdifConfig shall i use this?

Reply:
Yes that would be proper style.